libs/wingsdk/test/core/plugin-manager.test.ts

```diff
@@ -0,0 +1,171 @@
+import fs from "fs";
+import { join } from "path";
+import * as cdktf from "cdktf";
+import { PluginManager } from "../../src/core/plugin-manager";
+import * as tfaws from "../../src/target-tf-aws";
+import { mkdtemp } from "../../src/util";
+import { tfResourcesOfCount } from "../util";
+
+const PLUGIN_CODE = `
+var s3_bucket = require("@cdktf/provider-aws/lib/s3-bucket");
+
+exports.preSynth = function(app) {
+  new s3_bucket.S3Bucket(app, "PluginBucket", {bucket: "plugin-bucket"})  
+};
+
+exports.postSynth = function(config) {
+  const buckets = Object.keys(config.resource.aws_s3_bucket);
+  for (const bucket of buckets) {
+    config.resource.aws_s3_bucket[bucket].tags = { "plugin-tag": "plugin-value" };
+  }
+
+  return config;
+};
+
+exports.validate = function(config) {
+  throw new Error("some validation error");
+};
+`;
+
+const CREATE_S3_BUCKET_PLUGIN_CODE = `
+exports.postSynth = function(config) {
+  config.resource = config.resource || {};
+  config.resource.aws_s3_bucket = {
+    "plugin-bucket": {
+      "bucket": "plugin-bucket",
+    }
+  }
+  return config;
+};
+`;
+
+const DELETE_S3_BUCKET_PLUGIN_CODE = `
+exports.postSynth = function(config) {
+  config.resource.aws_s3_bucket = {};
+  return config;
+};
+`;
+
+const RETURN_NULL_CONFIG = `
+exports.postSynth = function(config) {
+  // no return
+};
+`;
+
+test("preSynth can add resources to construct tree", () => {
+  // GIVEN
+  const tmpDir = mkdtemp();
+  const pluginFile = join(tmpDir, "plugin.js");
+  fs.writeFileSync(pluginFile, PLUGIN_CODE);
+
+  // WHEN
+  const app = new tfaws.App({ outdir: tmpDir }); // exclude plugins from constructor for control
+  const pm = new PluginManager([pluginFile]);
+  pm.preSynth(app);
+  const output = app.synth();
+
+  // THEN
+  expect(tfResourcesOfCount(output, "aws_s3_bucket")).toEqual(1);
+  expect(
+    cdktf.Testing.toHaveResourceWithProperties(output, "aws_s3_bucket", {
+      bucket: "plugin-bucket",
+    })
+  ).toEqual(true);
+});
+
+test("postSynth can modify terraform config", () => {
+  // GIVEN
+  const tmpDir = mkdtemp();
+  const pluginFile = join(tmpDir, "plugin.js");
+  fs.writeFileSync(pluginFile, PLUGIN_CODE);
+
+  // WHEN
+  const app = new tfaws.App({ outdir: tmpDir }); // exclude plugins from constructor for control
+  const pm = new PluginManager([pluginFile]);
+  new tfaws.Bucket(app, "Bucket", {});
+
+  const synthOutput = app.synth();
+  const synthesizedStackPath = `${app.outdir}/main.tf.json`;
```

Comment:
How about we make this path a constant on `CdktfApp`?
```suggestion
  const synthesizedStackPath = app.terraformManifestPath;
```